include/faabric/scheduler/MpiWorld.h

```diff
@@ -3,19 +3,41 @@
 #include <faabric/mpi/mpi.h>
 
 #include <faabric/proto/faabric.pb.h>
-#include <faabric/scheduler/FunctionCallClient.h>
 #include <faabric/scheduler/InMemoryMessageQueue.h>
-#include <faabric/scheduler/MpiThreadPool.h>
-#include <faabric/state/StateKeyValue.h>
+#include <faabric/transport/MpiMessageEndpoint.h>
+#include <faabric/util/logging.h>
+#include <faabric/util/timing.h>
 
 #include <atomic>
-#include <thread>
+#include <list>
 
 namespace faabric::scheduler {
 typedef faabric::util::Queue<std::shared_ptr<faabric::MPIMessage>>
   InMemoryMpiQueue;
 
-std::string getWorldStateKey(int worldId);
+/* The untracked message buffer (UMB) keeps track of the asyncrhonous
```

Comment:
Typo, "untracked" vs "unacked". The object doesn't seem to have any properties that are unique to the messages being unacked, could we just call it a `MPIMessageBuffer` and put it in a separate file with some tests (see my comments below about encapsulating the logic from elsewhere in this file into this class).